standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2041,16 +2041,16 @@ public void drop_dataconnector(final String dcName, boolean ifNotExists, boolean
     Exception ex = null;
     RawStore ms = getMS();
     try {
-      ms.openTransaction();
       connector = getMS().getDataConnector(dcName);
-
-      if (connector == null) {
-        if (!ifNotExists) {
-          throw new NoSuchObjectException("DataConnector " + dcName + " doesn't exist");
-        } else {
-          return;
-        }
+    } catch (NoSuchObjectException e) {
```

Comment:
With this change we don't require [TestFilterHooks.java](https://github.com/apache/hive/pull/3304/files#diff-568151a80b62a98c19409989a18c38d8f90f71fddbf06286641821228f75dd9d) L228 and L231 anymore right?

Reply:
Actually, we don't need to catch NoSuchObjectException any more after this change. But we still need these two lines L229 and L230 because each test will create data connector and we should ensure data connector is dropped before each creating. Database test is also doing same thing.
https://github.com/apache/hive/blob/6b1a876c3df7db54553e95552bcc2c99eccf8ae2/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestFilterHooks.java#L204-L212